### PR TITLE
chore: babel, expect, zip bundles

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: packages/playwright-core/src/server/registry/browserFetcher.ts [hunk 1]
```diff
@@ -24,6 +23,8 @@ import { existsAsync } from '../../utils/fileUtils';
 import { debugLogger } from '../../common/debugLogger';
 import { download } from './download';
 
+const extract: typeof import('../../../bundles/zip/node_modules/extract-zip') = require('../../zipBundle').extract;
```
Comment (dgozman) [inline packages/playwright-core/src/server/registry/browserFetcher.ts:26]: Why not `typeof import('extract-zip')` similar to `yazl`?

File: packages/playwright-test/.npmignore [hunk 2]
```diff
@@ -1 +1,2 @@
-src
\ No newline at end of file
+src
```
Comment (dgozman) [inline packages/playwright-test/.npmignore:1]: We should switch this file to allowlist!

File: package.json [hunk 3]
```diff
@@ -45,6 +45,12 @@
   ],
   "devDependencies": {
     "@babel/cli": "^7.16.8",
+    "@babel/plugin-proposal-class-properties": "^7.16.7",
+    "@babel/plugin-proposal-export-namespace-from": "^7.16.7",
+    "@babel/plugin-proposal-logical-assignment-operators": "^7.16.7",
+    "@babel/plugin-proposal-nullish-coalescing-operator": "^7.16.7",
+    "@babel/plugin-proposal-optional-chaining": "^7.16.7",
+    "@babel/plugin-transform-modules-commonjs": "^7.17.9",
```
Comment (dgozman) [inline package.json:53]: Why do we have some of these modules here? Is it to build our own source?

File: packages/playwright-test/src/util.ts [hunk 4]
```diff
@@ -57,7 +57,6 @@ function filterStackTrace(e: Error) {
         return true;
       return !fileName.startsWith(PLAYWRIGHT_TEST_PATH) &&
              !fileName.startsWith(PLAYWRIGHT_CORE_PATH) &&
-             !fileName.startsWith(EXPECT_PATH) &&
```
Comment (dgozman) [inline packages/playwright-test/src/util.ts:60]: Fancy!

File: utils/generate_third_party_notice.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+This project incorporates components from the projects listed below. The original copyright notices and the licenses under which Microsoft received such components are set forth below. Microsoft reserves all rights not expressly granted herein, whether by implication, estoppel or otherwise.
+`);
+
+    const allPackages = {};
+    const projectDir = path.join(__dirname, '..', 'packages', project);
+    const bundlesDir = path.join(projectDir, 'bundles');
+    for (const bundle of fs.readdirSync(bundlesDir)) {
+      const dir = path.join(bundlesDir, bundle);
+      execSync('npm ci', { cwd: dir });
```
Comment (dgozman) [inline utils/generate_third_party_notice.js:37]: Why this one? I see every bundle package.json has `npm ci` in every command.

File: packages/playwright-core/src/server/registry/browserFetcher.ts (hunk 1 above)
Comment (pavelfeldman) [reply packages/playwright-core/src/server/registry/browserFetcher.ts:26]: extract-zip does not have a definitely-typed module, so i'd need a code devDependency in package.json

File: packages/playwright-test/.npmignore (hunk 2 above)
Comment (pavelfeldman) [reply packages/playwright-test/.npmignore:1]: Yeah, in a separate patch

File: package.json (hunk 3 above)
Comment (pavelfeldman) [reply package.json:53]: Yes, our build has a babel step.

File: utils/generate_third_party_notice.js (hunk 5 above)
Comment (pavelfeldman) [reply utils/generate_third_party_notice.js:37]: It was so that i could just run that one alone at any moment

**Pull request merged**